cxx-qt-gen/src/extract.rs

```diff
@@ -828,278 +825,63 @@ pub fn extract_qobject(
     module: &ItemMod,
     cpp_namespace_prefix: &[&str],
 ) -> Result<QObject, TokenStream> {
+    // Build a parser for the given ItemMod
+    //
+    // TODO: in the future steps from this extract.rs file will be moved into module parts
+    // of Parser
+    let mut parser = Parser::from(module.to_owned()).map_err(|err| err.to_compile_error())?;
+
+    // TODO: for now we only support one QObject per ItemMod block
+    // so extract the first qobject we find
+    if parser.cxx_qt_data.qobjects.len() != 1 {
+        return Err(Error::new(
+            module.span(),
+            "Only one QObject is currently supported in the ItemMod.",
+        )
+        .to_compile_error());
+    }
+    let (_, qobject) = parser.cxx_qt_data.qobjects.drain().take(1).next().unwrap();
+
     // Find the items from the module
     let original_mod = module.to_owned();
-    let items = &mut module
-        .to_owned()
-        .content
-        .expect("Incorrect module format encountered.")
-        .1;
 
     // Prepare variables to store struct, invokables, and other data
     //
     // The original Data Item::Struct if one is found
-    let mut original_data_struct = None;
+    let original_data_struct = qobject.data_struct;
     // The original RustObj Item::Struct if one is found
-    let mut original_rust_struct = None;
+    let original_rust_struct = qobject.rust_struct;
     // The name of the Qt object we are creating
     let qt_ident = quote::format_ident!("{}", original_mod.ident.to_string().to_case(Case::Pascal));
 
+    // Extract the normal and invokables from the Parser methods
+    let extracted = extract_invokables(&qobject.methods, cpp_namespace_prefix, &qt_ident)?;
+
     // A list of the invokables for the struct
-    let mut object_invokables = vec![];
+    let object_invokables = extracted.invokables;
     // A list of the normal methods (i.e. not invokables) for the struct
-    let mut object_normal_methods = vec![];
-    // A list of the rust only methods for the struct
-    let mut object_rust_methods = vec![];
-    // A list of original trait impls for the struct (eg `impl Default for Struct`)
-    let mut original_trait_impls = vec![];
+    let object_normal_methods = extracted.normal_methods;
```

Comment:
I presume by "normal" you mean not Q_INVOKABLE? If so, I think "non_invokable" would be more self-explanatory than "normal".

Reply:
These will eventually become properties implemented on the C++ object in Rust (but not exposed to C++), so it will mean that these methods will be in the context of C++ (eg self will be the C++ object) which is the same as the invokables, but allows you to have Rust only methods while being able to access C++ state (eg for calling signals etc).

Not sure what the best thing is to call them yet, they have been shoehorned into the old generation as "normal" methods :thinking:  Maybe `cpp_context_methods` or something?

Reply:
Alternatively we could call them simply "methods" and "invokable methods".

Reply:
Yup, probably best for now :-) As at least the rust_methods have gone now and are just passed straight across, so that helps a bit with the naming :')

Reply:
Okay let's go with just "methods" and "invokable methods"

Reply:
Yup changing to that

Reply:
Although "methods" and "invokables" might be simplest, as that's going to cause a lot of noise in the code we are about to remove to change to invokable_methods :-)

Reply:
I like "methods" and "invokables"